tools/permanence-calculator/components/controls/curve.js

```diff
@@ -61,12 +59,54 @@ const Curve = ({
     return () => {
       window.removeEventListener('resize', listener)
     }
-  }, [theme])
+  }, [theme, isVariable, value])
 
   const format = (value) => {
-    return `$${value.toFixed(0)}`
+    return `$${value}`
+  }
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    updateParamValueFromInput()
+  }
+
+  const updateParamValueFromInput = () => {
+    let v = parseInt(displayValue)
+    if (!isNaN(v)) {
+      if (v < 0) {
+        v = 0
+      }
+      if (v > 9999) {
+        v = 9999
+      }
+      setValue([
+        [0, v],
+        [20, v],
+        [40, v],
+        [60, v],
+        [80, v],
+        [100, v],
+      ])
+      setDisplayValue(v)
+      if (!isVariable && chart) chart.update(v)
+    } else {
+      setDisplayValue(parseInt(value[0][1]))
+    }
   }
 
+  const updateParamDisplayValue = (e) => {
+    let normalized = e.target.value.replace('$', '')
```

Comment:
small UI nit: it might be worth replacing any non-digit character here (this would allow us to support manually entry of "$7,000" for example)

Reply:
Good flag. I do think we want decimals though. But we could probably be ok with everything else!